modules/watcher/watcher.go

```diff
@@ -76,6 +76,10 @@ func run(ctx context.Context, desc string, opts *CreateWatcherOpts) {
 				return
 			}
 			log.Debug("Watched file for %s had event: %v", desc, event)
+			if event.Op == fsnotify.Chmod {
+				_ = watcher.Close()
+				return
+			}
```

Comment:
Is that really the correct error handling here?
To me, this looks like killing the watcher as soon as you encounter a `chmod` event.
And not the intended no-op?

Reply:
I agree with @delvh, maybe a continue statement or something similar may be more appropriate.

In addition, it may be preferable to add it before the log statement so the log doesn't fill up the log  in case someone does many chmod commands

Reply:
```suggestion
			if event.Op != fsnotify.Chmod {
				_ = watcher.Close()
				return
			}
```

Reply:
Or:
```suggestion
			if event.Op == fsnotify.Chmod {
				continue
			}
```